File: PlantCareBuddy/PlantCareBuddy.Application/DTOs/Dashboard/DashboardOverviewDto.cs
namespace PlantCareBuddy.Application.DTOs.Dashboard
{
    public class DashboardOverviewDto
    {
        public List<PlantNeedsAttentionDto> PlantsNeedingAttention { get; set; }
        public List<UpcomingCareDto> UpcomingCareTasks { get; set; }
        public DashboardStatsDto Stats { get; set; }
    }
}

File: client/src/models/Dashboard/dashboardOverviewDto.ts
import { DashboardStatsDto } from "./dashboardStatsDto";
import { PlantNeedsAttentionDto } from "./plantNeedsAttentionDto";
import { UpcomingCareDto } from "./upcomingCareDto";

export interface DashboardOverviewDto {
    plantsNeedingAttention: PlantNeedsAttentionDto[];
    upcomingCareTasks: UpcomingCareDto[];
    stats: DashboardStatsDto;
}